### PR TITLE
#164771605 Implements Fastlane Integration for Slack

### PR DESCRIPTION
**Pull request by Maxfurry**
### What does this PR do?
It updates Grade and implements Fastlane integration

### Description of Task to be completed?
* Install and setup Fastlane
* Setup environment variable with slack webhook and token
* Succesful send a message to Slack channel using fastlane

### How should this be manually tested?
* Clone branch
* Checkout into this branch ch-fastlane-integration-164771605
* Get webhook for Converge Codelab on slack
* Setup SLACK_TOKEN as environment variable on your system
* Run fastlane deploy in the terminal

### What are the relevant pivotal tracker stories?
[#164771605](https://www.pivotaltracker.com/story/show/164771605)

### Screenshot
Screen Shot 2019-04-12 at 11 24 01 AM

### CODE REVIEW

**Comment by bibangamba** [comment]
please include a screenshot of the uploaded `.apk` file

**Review by bibangamba — CHANGES_REQUESTED** [review]
I couldn't see any lane that handles uploading the apk to the slack channel. please include this implementation as well.

**Comment by Maxfurry** [comment]
It is on line 37

File: fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  lane :deploy do
+    gradle(task: "clean assembleRelease")
+    slack_build()
+  end
+
+  desc "Upload new version to Slack Channel"
+  private_lane :slack_build do |options|
+    file_path = lane_context[SharedValues::GRADLE_APK_OUTPUT_PATH]
+    file_name = file_path.gsub(/\/.*\//,"")
+    token = ENV["SLACK_URL"]
```
Comment (bibangamba) [inline fastlane/Fastfile]: the token is different from the `webhook url`. please generate a token which will authorize your upload to slack

Comment (Maxfurry) [reply fastlane/Fastfile]: Noted

**Review by bibangamba — APPROVED** [review]
LGTM

**Pull request merged**